src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -102,23 +191,41 @@ public override int GetHashCode()
             return hash;
```

Comment:
How does this allocate? HashCodeCombiner is a struct. Is it on any of the calls to Add? when it internally calls s.GetHashCode() or similar?